pandas/core/groupby/generic.py

```diff
@@ -1056,9 +1058,47 @@ def _cython_agg_blocks(
             result = no_result
             locs = block.mgr_locs.as_array
             try:
-                result, _ = self.grouper.aggregate(
-                    block.values, how, axis=1, min_count=min_count
+
+                cat_method_blacklist = (
```

Comment:
I copied this list from `asv_bench.benchmarks.groupby.method_blacklist['object']` and appended the "add" method. Is there a better way of blacklisting these methods which shouldn't be applied to categorical codes?

`tests.groupby.test_function.test_arg_passthru` is an example of a test failure without this blacklisting.